### ISSUE TITLE
Escape character not properly escaped in JSON output

### ISSUE THREAD

**Issue by nicolaslt** [opener-project/kaf2json#2]
Hi,

It seems the string returned by kaf2json is not valid JSON if the original text contains backslashes.

Here's an example:
```bash
echo "test \\ this test in english" | language-identifier | tokenizer | kaf2json
```

returns:
```json
{
        "text"      : "test \ this test in english ",
        "language"  : "en",
        "terms"     : {},


        "lp"        : [
{
        "name"     :"opener-sentence-splitter-en",
        "timestamp":"2016-01-18T15:53:51Z",
        "version"  :"0.0.1",
        "layer"    :"text"
},
{
        "name"     :"opener-tokenizer-en",
        "timestamp":"2016-01-18T15:53:51Z",
        "version"  :"1.0.1",
        "layer"    :"text"
}]
}
```
That is invalid JSON. The valid version of it would be:
```json
{
        "text"      : "test \\ this test in english ",
        "language"  : "en",
        "terms"     : {},


        "lp"        : [
{
        "name"     :"opener-sentence-splitter-en",
        "timestamp":"2016-01-18T15:53:51Z",
        "version"  :"0.0.1",
        "layer"    :"text"
},
{
        "name"     :"opener-tokenizer-en",
        "timestamp":"2016-01-18T15:53:51Z",
        "version"  :"1.0.1",
        "layer"    :"text"
}]
}

```

That's on kaf2json v2.0.0

**Comment by giannismelidis** [comment]
Hi Nicolas,

Thanks for using OpeNER!

This is not invalid JSON:
```
{
    "text": "test \ this test in english ",
    "language": "en",
    "terms": {},


    "lp": [{
        "name": "opener-sentence-splitter-en",
        "timestamp": "2016-01-18T16:28:59Z",
        "version": "0.0.1",
        "layer": "text"
    }, {
        "name": "opener-tokenizer-en",
        "timestamp": "2016-01-18T16:28:59Z",
        "version": "1.0.1",
        "layer": "text"
    }]
}
```

You can try it here: http://jsonlint.com/

Best,
Giannis Melidis | Software Engineer @ Olery B.V.

> nicolaslt <mailto:notifications@github.com>
> 18 Jan 2016 16:56
>
> Hi,
>
> It seems the string returned by kaf2json is not valid JSON if the
> original text contains backslashes
>
> Here's an example:
>
> echo "test \\ this test in english" | language-identifier | tokenizer
> | kaf2json
>
> returns:
>
> {
> "text" : "test \ this test in english ",
> "language" : "en",
> "terms" : {},
>
>
> "lp" : [
> {
> "name" :"opener-sentence-splitter-en",
> "timestamp":"2016-01-18T15:53:51Z",
> "version" :"001",
> "layer" :"text"
> },
> {
> "name" :"opener-tokenizer-en",
> "timestamp":"2016-01-18T15:53:51Z",
> "version" :"101",
> "layer" :"text"
> }]
> }
>
> That is invalid JSON The valid version of it would be:
>
> {
> "text" : "test \\ this test in english ",
> "language" : "en",
> "terms" : {},
>
>
> "lp" : [
> {
> "name" :"opener-sentence-splitter-en",
> "timestamp":"2016-01-18T15:53:51Z",
> "version" :"001",
> "layer" :"text"
> },
> {
> "name" :"opener-tokenizer-en",
> "timestamp":"2016-01-18T15:53:51Z",
> "version" :"101",
> "layer" :"text"
> }]
> }
>
> That's on kaf2json v200
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opener-project/kaf2json/issues/2>.
>

-- 
Giannis Melidis
Software Engineer | Olery

**Comment by nicolaslt** [comment]
Hi,

I think jsonlint.com is wrong in this regard. The [original JSON spec](http://json.org/) is quite clear in that a "\" char may only be followed by one of the following characters 
```
\/bfnrt
```
or "u" followed by 4 hexa digits to specify a unicode char.

A short list of things I know of that don't accept this string as valid JSON: 
  - [jq](https://stedolan.github.io/jq/)
  - the default python parser (python -mjson.tool)
  - the default javascript parser (JSON.parse)
  - [this online validator](https://jsonformatter.curiousconcept.com/). Regardless of which RFC is selected

**Comment by giannismelidis** [comment]
This is standard ruby behaviour that's why I got confused. The problem was with the XSLT template which you have to tell explicitly to escape the backslashes. Fixed in version 2.0.2.
